File: app/src/main/java/pt/ipbeja/pi/piproject/persistence/Converters.java
package pt.ipbeja.pi.piproject.persistence;

import androidx.room.TypeConverter;

import java.util.Date;

/**
 * Converts Date into int timestamps back and forth
 */
public class Converters {
    @TypeConverter
    public Date fromTimestamp(Long value) {
        return value == null ? null : new Date(value);
    }

    @TypeConverter
    public Long dateToTimestamp(Date date) {
        if (date == null) {
            return null;
        } else {
            return date.getTime();
        }
    }
}

File: app/src/main/java/pt/ipbeja/pi/piproject/persistence/Converters.kt
package pt.ipbeja.pi.piproject.persistence

import androidx.room.TypeConverter
import java.util.Date

/**
 * Converts Date into int timestamps back and forth
 */
class Converters {
    @TypeConverter
    fun fromTimestamp(value: Long?): Date? {
        return if (value == null) null else Date(value)
    }

    @TypeConverter
    fun dateToTimestamp(date: Date?): Long? {
        return date?.time
    }
}